Plugin/Connector/Job/AttributePlugin.php

```diff
@@ -33,33 +33,49 @@ public function afterAddAttributes(Attribute $attributeJob): void
     {
         if (!$this->configHelper->isReferenceEntitiesEnabled()) {
             $connection = $this->entitiesHelper->getConnection();
-            $tmpTable = $this->entitiesHelper->getTableName('attribute');
             $entityTable = $this->entitiesHelper->getTable('akeneo_connector_entities');
-            $select = $connection->select()->from($tmpTable)
-                ->where("frontend_input = 'smile_custom_entity'");
+            $select = $connection->select()
+                ->from(
+                    ['tmp' => $this->entitiesHelper->getTableName('attribute')],
+                    [
+                        '_entity_id',
+                        'reference_data_name',
+                        'frontend_input',
+                    ]
+                )->joinLeft(
+                    ['eav' => $this->entitiesHelper->getTable('eav_attribute')],
+                    'tmp._entity_id = eav.attribute_id',
+                    [
+                        'magento_frontend_input' => 'frontend_input',
+                    ]
+                )->where("tmp.frontend_input = 'smile_custom_entity'");
             $attributes = $connection->fetchAll($select);
             $productEntityTypeId = $this->eavSetup->getEntityTypeId(ProductAttributeInterface::ENTITY_TYPE_CODE);
             foreach ($attributes as $attribute) {
-                if ($attribute['reference_data_name']) {
-                    $entitySelect = $connection->select()->from($entityTable, ['entity_id'])
-                        ->where('code = ?', $attribute['reference_data_name'])
-                        ->where('import = "smile_custom_entity"');
-                    $customEntityTypeId = $connection->fetchOne($entitySelect);
-                    if (!$customEntityTypeId) {
-                        $customEntityTypeId = $this->eavSetup->getDefaultAttributeSetId(
-                            CustomEntityInterface::ENTITY
-                        );
-                    }
-                    $values = [
-                        'is_html_allowed_on_front' => 1,
-                        'custom_entity_attribute_set_id' => $customEntityTypeId,
-                    ];
-                    $this->eavSetup->updateAttribute(
-                        $productEntityTypeId,
-                        $attribute['_entity_id'],
-                        $values
+                if (
+                    !$attribute['reference_data_name'] || !$attribute['magento_frontend_input']
+                    || $attribute['frontend_input'] !== $attribute['magento_frontend_input']
+                ) {
+                    continue;
```

Comment:
What happen to the attribut that have a reference data name and the wrong frontend_input ? They will just be imported as normal attribut ?
We might need to add a warning in the import log to inform the user that the behavior of the attribute will might not be the one expected ?

Reply:
These attributes will not be updated/created by akeneo native import if we have a frontend_input mismatch (and we should skip that too).  A message will be added to the import log:
 `'The attribute %1 was skipped because its type is not the same between Akeneo and Adobe Commerce/Magento. Please delete it in Adobe Commerce/Magento and try a new import' `